code/modules/mob/living/silicon/laws.dm

```diff
@@ -1,7 +1,93 @@
+#define MAX_INTEGRITY 100
+#define MIN_INTEGRITY -100
+#define MODE_FACTORY "factory-default"
+#define MODE_FIRMWARE "load-firmware-laws"
+#define MODE_LISP "lawchange-induced-synthetic-psychosis"
+#define MODE_REGULAR "regular"
+#define THRESHOLD_LISP -50
+#define THRESHOLD_FACTORY 80
+
 /mob/living/silicon
 	var/datum/ai_laws/laws = null
 	var/list/additional_law_channels = list("State" = "")
-	var/lawcooldown = FALSE
+	var/integrity = MAX_INTEGRITY
+	var/law_mode = MODE_REGULAR
+
+/mob/living/silicon/proc/adjust_integrity(amount=0)
+	var/old_integrity = clamp(integrity, MIN_INTEGRITY, MAX_INTEGRITY)
+	var/new_integrity = clamp((integrity + amount), MIN_INTEGRITY, MAX_INTEGRITY)
+
+	// if we're in regular mode
+	if(law_mode == MODE_REGULAR)
+		// adjust integrity to the new value
+		integrity = new_integrity
+		// if integrity falls below the LISP threshold
+		if(integrity <= THRESHOLD_LISP)
+			// we enter LISP mode
+			do_mode(MODE_LISP)
+
+	// otherwise, if we're in factory default mode
+	else if(law_mode == MODE_FACTORY)
+		// adjust integrity to the new value
+		integrity = new_integrity
+		// if integrity falls below the LISP threshold
+		if(integrity <= THRESHOLD_LISP)
+			// we enter LISP mode
+			do_mode(MODE_LISP)
+		// otherwise if integrity rises to the maximum value
+		else if(integrity >= MAX_INTEGRITY)
+			// we enter regular mode; reload from firmware
+			do_mode(MODE_FIRMWARE)
+
+	// otherwise, if we're in lisp mode
+	else if(law_mode == MODE_LISP)
+		// when in LISP, integrity can only go up
+		integrity = max(old_integrity, new_integrity)
+		// if integrity rises above the factory default threshold
+		if(integrity >= THRESHOLD_FACTORY)
+			// we enter factory default mode
+			do_mode(MODE_FACTORY)
+
+/mob/living/silicon/proc/do_mode(new_mode=MODE_REGULAR)
```

Comment:
Nitpick but, `do_mode` sounds odd to me. I think something more descriptive like `change_law_mode` would be better.